### ISSUE TITLE
No routes advertised to BGP peers for bgp-graceful-restart-deferral-time if graceful-restart is enabled

### ISSUE THREAD

**Issue by camrossi** [cilium/cilium#30367]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I have Cilium configured for eBGP to external switches.

As per Cilium [Documentation](https://docs.cilium.io/en/latest/network/bgp-control-plane/#graceful-restart) `When enabled, graceful restart capability is advertised for IPv4 and IPv6 address families.` 
This however this is problematic, my routers are configured only for v4 and this leads to no routes being advertised by Cilium/goBGP for bgp-graceful-restart-deferral-time that is 300s. This is expected as per the RFC as Cilium never gets the  End-of-RIB marker for the v6 AF... but this is never gonna happens as my routers are v4 only.

This is the exact same issue I faced with Kube-Router:
https://github.com/cloudnativelabs/kube-router/issues/1323

There is also an issue open for GoBGP:
https://github.com/osrg/gobgp/issues/2524

Could it be considered to allow Cilium to send the GR capability only for the enabled AF in the `CiliumBGPPeeringPolicy`as was done for kube-router?

Also even if I configure my routers with dual stack the issue is still there as Cilium is advertising the v4/v6 GR capability to the V4 peer and v4/v6 GR capability to the v6 peer.

Even if is the same physical device the BGP process is not the same so still same issue of missing End-of-RIB marker

### Cilium Version

root@talos-62e-tvr:/home/cilium# cilium version
Client: 1.15.0-rc.1 f582c55c 2024-01-16T13:41:43+01:00 go version go1.21.6 linux/amd64
Daemon: 1.15.0-rc.1 f582c55c 2024-01-16T13:41:43+01:00 go version go1.21.6 linux/amd64

### Kernel Version

Linux talos-62e-tvr 6.1.73-talos #1 SMP PREEMPT_DYNAMIC Wed Jan 17 14:54:51 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

1.29.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by harsimran-pabla** [comment]
Hi @camrossi, Can you try enabling just ipv4 unicast family in cilium, by default both ipv4 and ipv6 unicast are enabled. 

```  
virtualRouters:
  - localASN: 65001
    exportPodCIDR: true
    neighbors:
    - peerAddress: "172.0.0.1/32"
      peerASN: 65000
      gracefulRestart:
        enabled: true
        restartTimeSeconds: 120
      families:
      - afi: ipv4
        safi: unicast
```

**Comment by camrossi** [comment]
Hi @harsimran-pabla !

That did it.
I tested this with v4 only routers and a DualStack setup as well and it works as expected. Cilium advertise the routes right away after the BGP process comes up. 

The v6 peer config is the same just with `afi: ipv6`

Questions:

1) Can we document this options in the BGP config guide? I do see these documented in the [Fields](https://docs.cilium.io/en/stable/network/bgp-control-plane/#fields) section. 
2) We should probably explain this issue in the [graceful-restart](https://docs.cilium.io/en/stable/network/bgp-control-plane/#graceful-restart) section, pretty much I think this is something most configs will have to address.
3) Is there any plan to have goBGP auto-detect this? 

Anyway for now I have a working config so that is already amazing! 
Thank you!

**Comment by YutaroHayakawa** [comment]
Document PR: https://github.com/cilium/cilium/pull/30455
GoBGP PR: https://github.com/osrg/gobgp/pull/2761

**Comment by YutaroHayakawa** [comment]
GoBGP was upgraded in https://github.com/cilium/cilium/pull/30626. Now the main branch should be fixed.

**Comment by YutaroHayakawa** [comment]
Now, the fix is applied on the main, v1.14, and v1.15 branches. They will be available from the next release (next month). Closing.

**Comment by Founderbn2014** [comment]
> Now, the fix is applied on the main, v1.14, and v1.15 branches. They will be available from the next release (next month). Closing.

i had the same problem，v1.15.1 don't seem to be working。
